bip-0078.mediawiki

```diff
@@ -113,18 +116,23 @@ The payjoin proposal MUST:
 * Only fill the <code>witnessUTXO</code> or <code>nonWitnessUTXO</code> for the additional inputs.
 
 The payjoin proposal MAY:
-* Add, remove or modify the outputs belonging to the receiver.
+* Add, or replace the outputs belonging to the receiver unless output substitution is disabled.
 
 The payjoin proposal MUST NOT:
 * Shuffle the order of inputs or outputs, the additional outputs or additional inputs must be inserted at a random index.
 * Decrease the absolute fee of the original transaction.
 
+The payjoin proposal SHOULD NOT:
+* Include mixed input types until May 2022. Mixed inputs were previously completely disallowed so this gives some grace period for senders to update.
+
```

Comment:
> @NicolasDorier: I believe we don't need any grace period or anything
> if the receiver add a mixed input, then the sender might not like it and thus not use Payjoin and fallback
> which is fine
> we should just notify libraries

Applies also to lines 106-108

Reply:
Not sure where you're quoting him from, anyway, that's why there's `SHOULD NOT`, and not `MUST NOT`. It's simply to increase compatibility to prevent a bunch of PayJoins failing in the meantime.